meilisearch/client.py

```diff
@@ -235,6 +254,90 @@ def get_keys(self) -> Dict[str, str]:
         """
         return self.http.get(self.config.paths.keys)
 
+    def create_key(
+        self,
+        options: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, int]:
+        """Creates a new API key.
+
+        Parameters
+        ----------
+        options:
+            Options, the information to use in creating the key (ex: { 'actions': ['*'], 'indexes': ['movies'], 'description': 'Search Key', 'expiresAt': '22-01-01' }).
+            An `actions`, an `indexes` and a `expiresAt` fields are mandatory,`None` should be specified for no expiration date.
+            `actions`: A list of actions permitted for the key. ["*"] for all actions.
+            `indexes`: A list of indexes permitted for the key. ["*"] for all indexes.
+            Note that if an expires_at value is included it should be in UTC time.
+
+        Returns
+        -------
+        keys:
+            The new API key.
+            https://docs.meilisearch.com/reference/api/keys.html#get-keys
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        if options is None:
```

Comment:
If it is correct that options cannot be `None` this would change to:

```py
if not options.get("expiresAt"):
    options["expiresAt"] = None

return self.http.post(f'{self.config.path.keys)', options)
```

I'm not sure if MeiliSearch expects the `expiresAt` key to be present if it is not set. If not the `if options.get('expiresAt')...` part could be removed.

Reply:
For security reasons`Meilisearch` force the user to specify `expiresAt ` at `null` if he doesn't want an expiration date.

Reply:
If you make the update I suggested it will add the `expiresAt` key as being `null` to sent in the request if they don't include it. So I think either way could work. Basically you can either take the routing of assuming not including it means they want it to be `null` or you could force them to explicitly include it as `None` and raise an exception if it isn't there. I can't think of any big advantages of one way over the other so either option seems good.

Because `options` is a dictionary the one that that might be helpful, with either option, is to include the expected keys in the doc string. Also maybe valid the keys are correct.

The only reason I didn't do this in my update is because the data is being passed through a Pydantic class so it is already validated before it makes it to the method.

Reply:
Thank you for your advice! I think it is better than the user is obliged to put an expiration date since this is how Meilisearch works. I remove the `if options is None:`